Add API method to rename devices

thomseddon opened the pull request:
Not sure on how you'd like this method to be presented? Let me know if you'd prefer the new hostname to be in the body (I just wanted some kind of differentiation).

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/d1adcb9a-2614-4831-beaa-7b38ad251267) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/d1adcb9a-2614-4831-beaa-7b38ad251267) completed: **86 new issues, 58 updated code elements**

murrant commented on html/includes/api_functions.inc.php:
```diff
@@ -1262,6 +1262,41 @@ function update_device()
+            $code = 200;
+            $message = 'Device has been renamed';
+        } else {
+            $message = 'Device failed to be reanmed';
```
small typo here `reanmed` -> `renamed`

laf commented:
Thanks @thomseddon.

The API code went through a little refactor the other week so you'll need to update the code a bit to match (I can do that if you haven't got time?)

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/db1f8ebf-1dab-40b0-9335-05e72c4fd85c) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/01b3b3df-fbd6-42e3-8721-736bddcd011c) was created.

thomseddon commented:
Fixed the typo, thanks for spotting that.

I've also added `check_is_admin();` to match `update_device`?

@laf what branch should I rebase from? I've just checked and it seems up to date with master on here, just let me know what I'm missing, can get it fixed today :)

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/db1f8ebf-1dab-40b0-9335-05e72c4fd85c) completed: **1 updated code elements**

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/01b3b3df-fbd6-42e3-8721-736bddcd011c) completed: **1 updated code elements**

laf commented:
> @laf what branch should I rebase from? I've just checked and it seems up to date with master on here, just let me know what I'm missing, can get it fixed today :)

@thomseddon sorry for the delay, busy week as you know.

So sending an error back can be done like: https://github.com/librenms/librenms/blob/master/html/includes/api_functions.inc.php#L255

Sending the end response can be done like: https://github.com/librenms/librenms/blob/master/html/includes/api_functions.inc.php#L263

Happy to update if you don't have time.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/21cb6495-07c1-40bb-974e-901e433f4260) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/f71eb502-8bfa-4968-a4ed-fa1528833c09) was created.

thomseddon commented:
@laf ah sorry, I see it now - my mistake!

I've switched to the new functions to form the responses (much nicer :+1:) and tided up a couple of other things (erroneous new line, unnecessary full stop in response)

Happy to make any more amends :)

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/21cb6495-07c1-40bb-974e-901e433f4260) completed: **1 updated code elements**

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/f71eb502-8bfa-4968-a4ed-fa1528833c09) completed: **1 updated code elements**

laf commented:
Thanks for doing this @thomseddon :)
